File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapField.java
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation;

import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapFields;

import java.lang.annotation.*;

@Documented
@Target(ElementType.FIELD)
@Retention(RetentionPolicy.RUNTIME)
@Repeatable(WrapFields.class)
/**
 * Auto setup field
 */
public @interface WrapField {
    String mcpName();
    String targetMap();
    String customSrgName() default "none";
}

File: src/main/java/rbq/wtf/lycoris/client/wrapper/wrappers/annotation/WrapField.kt
package rbq.wtf.lycoris.client.wrapper.wrappers.annotation

import rbq.wtf.lycoris.client.wrapper.MapEnum
import rbq.wtf.lycoris.client.wrapper.wrappers.annotation.repeat.WrapFields

@MustBeDocumented
@Target(AnnotationTarget.FIELD)
@Retention
@JvmRepeatable(
    WrapFields::class
)
/**
 * Auto setup field
 */
annotation class WrapField(val mcpName: String, val targetMap: MapEnum, val customSrgName: String = "none")